Cli problems configuring token

```text
$ ./exercism configure -v --token=xxxxx
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/exercism/cli/debug.DumpRequest(0x90b4900)
	/Users/ekingery/dev/exercism/cli/debug/debug.go:50 +0x6c4
github.com/exercism/cli/api.(*Client).Do(0x90fde04, 0x90b4900, 0x3, 0x909a2c0, 0x1f)
	/Users/ekingery/dev/exercism/cli/api/client.go:66 +0x21
github.com/exercism/cli/api.(*Client).IsPingable(0x90fde04, 0x0, 0x0)
	/Users/ekingery/dev/exercism/cli/api/client.go:100 +0xf1
github.com/exercism/cli/cmd.runConfigure(0x84b688e, 0x5, 0x901e285, 0xe, 0x909a2a0, 0x1f, 0x84c483f, 0x1a, 0xffe20ed5, 0x8, ...)
	/Users/ekingery/dev/exercism/cli/cmd/configure.go:94 +0xa2b
github.com/exercism/cli/cmd.glob..func1(0x88d25e0, 0x9084420, 0x0, 0x2, 0x0, 0x0)
	/Users/ekingery/dev/exercism/cli/cmd/configure.go:43 +0xe3
github.com/spf13/cobra.(*Command).execute(0x88d25e0, 0x9084400, 0x2, 0x2, 0x88d25e0, 0x9084400)
	/Users/ekingery/golang/pkg/mod/github.com/spf13/cobra@v0.0.0-20170731170427-b26b538f6930/command.go:649 +0x2e4
github.com/spf13/cobra.(*Command).ExecuteC(0x88d2a60, 0x0, 0x0, 0x8440980)
	/Users/ekingery/golang/pkg/mod/github.com/spf13/cobra@v0.0.0-20170731170427-b26b538f6930/command.go:728 +0x21f
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/ekingery/golang/pkg/mod/github.com/spf13/cobra@v0.0.0-20170731170427-b26b538f6930/command.go:687
github.com/exercism/cli/cmd.Execute()
	/Users/ekingery/dev/exercism/cli/cmd/root.go:39 +0x23
main.main()
	/Users/ekingery/dev/exercism/cli/exercism/main.go:6 +0x17
```

--- a/debug/debug.go
+++ b/debug/debug.go
@@ -44,10 +44,10 @@ func DumpRequest(req *http.Request) {
 	body := io.TeeReader(req.Body, &bodyCopy)
 	req.Body = ioutil.NopCloser(body)
 
-	temp := req.Header.Get("Authorization")
+	authHeader := req.Header.Get("Authorization")
 
-	if !UnmaskAPIKey {
-		if token := strings.Split(temp, " ")[1]; token != "" {
+	if authParts := strings.Split(authHeader, " "); len(authParts) > 1 && !UnmaskAPIKey {
+		if token := authParts[1]; token != "" {
 			req.Header.Set("Authorization", "Bearer "+Redact(token))
 		}
 	}
@@ -62,7 +62,7 @@ func DumpRequest(req *http.Request) {
 	Println("========================= END DumpRequest =========================")
 	Println("")
 
-	req.Header.Set("Authorization", temp)
+	req.Header.Set("Authorization", authHeader)
 	req.Body = ioutil.NopCloser(&bodyCopy)
 }
 

--- a/debug/debug_test.go
+++ b/debug/debug_test.go
@@ -2,6 +2,7 @@ package debug
 
 import (
 	"bytes"
+	"net/http"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -27,6 +28,76 @@ func TestVerboseDisabled(t *testing.T) {
 	if b.String() != "" {
 		t.Error("expected '' got", b.String())
 	}
+}
+
+func TestDumpRequest(t *testing.T) {
+	testCases := []struct {
+		desc    string
+		auth    string
+		verbose bool
+		unmask  bool
+	}{
+		{
+			desc:    "Do not attempt to dump request if 'Verbose' is set to false",
+			auth:    "",
+			verbose: false,
+			unmask:  false,
+		},
+		{
+			desc:    "Dump request without authorization header",
+			auth:    "", //not set
+			verbose: true,
+			unmask:  false,
+		},
+		{
+			desc:    "Dump request with malformed 'Authorization' header",
+			auth:    "malformed",
+			verbose: true,
+			unmask:  true,
+		},
+		{
+			desc:    "Dump request with properly formed 'Authorization' header",
+			auth:    "Bearer abc12-345abcde1234-5abc12",
+			verbose: true,
+			unmask:  false,
+		},
+	}
+
+	b := &bytes.Buffer{}
+	output = b
+	for _, tc := range testCases {
+		Verbose = tc.verbose
+		UnmaskAPIKey = tc.unmask
+		r, _ := http.NewRequest("GET", "https://api.example.com/bogus", nil)
+		if tc.auth != "" {
+			r.Header.Set("Authorization", tc.auth)
+		}
+
+		DumpRequest(r)
+		if tc.verbose {
+			assert.Regexp(t, "GET /bogus", b.String(), tc.desc)
+			assert.Equal(t, tc.auth, r.Header.Get("Authorization"), tc.desc)
+			if tc.unmask {
+				assert.Regexp(t, "Authorization: "+tc.auth, b.String(), tc.desc)
+			}
+		} else {
+			assert.NotRegexp(t, "GET /bogus", b.String(), tc.desc)
+		}
+	}
+}
+
+func TestDumpResponse(t *testing.T) {
+	b := &bytes.Buffer{}
+	output = b
+	Verbose = true
+	r := &http.Response{
+		StatusCode: 200,
+		ProtoMajor: 1,
+		ProtoMinor: 1,
+	}
+
+	DumpResponse(r)
+	assert.Regexp(t, "HTTP/1.1 200 OK", b.String())
 }
 
 func TestRedact(t *testing.T) {

